libnopegl/src/path.h

```diff
@@ -27,6 +27,18 @@
 
 struct path;
 
+#define SEGMENT_FLAG_NEW_ORIGIN (1 << 0) /* the current segment does not overlap with the previous one */
+#define SEGMENT_FLAG_LINE       (1 << 1) /* the current segment is a simple line (not a curve) */
+
+struct path_segment {
+    float poly_x[4];
+    float poly_y[4];
+    float poly_z[4];
+    int step_start;
```

Comment:
nit: we might want to switch it to int32_t in a separate commit.

Reply:
This one is tricky: it needs to change all the `int` into `i32` in the `path` module, but technically those should be `size_t`. Unfortunately there is a bunch of arithmetic currently, and I'm a bit afraid to break something due to signed operations sometimes.